### PR TITLE
Add regression test for issue #3491: nullable enum with null in 'in' operator

### LINKED ISSUE #3491: Microsoft.OData.Core 8.4.3: in() with nullable enum and null throws The string 'null' is not a valid enumeration type constant.
After upgrading Microsoft.OData.Core from 8.4.2 to 8.4.3, queries using the in() operator on a nullable enum property fail when null is included in the value list.

This worked correctly in 8.4.2, but in 8.4.3 it throws:
`Microsoft.OData.ODataException: The string 'null' is not a valid enumeration type constant.`

Stack trace indicates the failure originates from enum constant validation logic inside:

` MetadataBindingUtils.cs `
The failure occurs specifically during:
`queryOptions.Validate(validationSettings);`

We assume this issue may be related to the recent flag enum support changes introduced in commit #3324

```
public enum Status
{
    New = 1,
    Active = 2,
    Closed = 3
}

public class Order
{
    public int Id { get; set; }

    // Nullable enum
    public Status? Status { get; set; }
}
```

Failing Url 
`GET https://localhost:5001/odata/Orders?$filter=Status in (null, 'Active')`
## Assemblies affected

*Microsoft.OData.Core 8.4.3*

### PR DESCRIPTION
**Pull request by xuzhg**
Verify that \=Status in (null, 'Active') works correctly for nullable enum properties. The issue was reported against 8.4.3 but is already fixed in dev-9.x.

### Issues

*This pull request fixes #3491.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### Repository notes

Team members can start a CI build by adding a comment with the text `/AzurePipelines run` to a PR. A bot may respond indicating that there is no pipeline associated with the pull request. This can be ignored if the build is triggered.

Team members should **not** trigger a build this way for pull requests coming from forked repositories. They should instead trigger the build manually by setting the "branch" to `refs/pull/{prId}/merge` where `{prId}` is the ID of the PR.

### CODE REVIEW

File: test/UnitTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs [hunk 1]
```diff
@@ -2675,6 +2675,50 @@ public void FilterWithInOperationWithEnumsInvalidMemberNames_ThrowsIsNotValidEnu
...
+
+            var model = new EdmModel();
+            model.AddElement(enumType);
+            model.AddElement(entityType);
+            model.AddElement(container);
+
+            // Act - parse "$filter=Status in (null, 'Active')"
+            FilterClause filter = new ODataQueryOptionParser(
+                model, entityType, entitySet,
+                new Dictionary<string, string> { { "$filter", "Status in (null, 'Active')" } }).ParseFilter();
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply test/UnitTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs:2703]: No need to call that. Leave it unchanged. Thanks for your suggestion.
